src/xrpld/app/tx/detail/Credentials.cpp

```diff
@@ -0,0 +1,486 @@
+        return ret;
+
+    auto const optSubj = ctx.tx[~sfSubject];
+    auto const optIss = ctx.tx[~sfIssuer];
+
+    if (!optSubj && !optIss)
+    {
+        // Neither field is present, the transaction is malformed.
+        JLOG(ctx.j.trace()) << "Malformed transaction: "
+                               "Invalid Subject and Issuer fields combination.";
+        return temMALFORMED;
+    }
+
+    // Make sure that the passed account is valid.
+    if ((optSubj && !AccountID(*optSubj)) || (optIss && !AccountID(*optIss)))
+    {
+        JLOG(ctx.j.trace()) << "Malformed transaction: Subject or Issuer "
+                               "field zeroed.";
+        return temINVALID_ACCOUNT_ID;
+    }
+
+    return preflight2(ctx);
+}
+
+TER
+CredentialDelete::preclaim(PreclaimContext const& ctx)
+{
+    AccountID const account{ctx.tx[sfAccount]};
+    auto const subject = ctx.tx[~sfSubject].value_or(account);
+    auto const issuer = ctx.tx[~sfIssuer].value_or(account);
+    auto const credType(ctx.tx[sfCredentialType]);
+
+    if (!ctx.view.exists(keylet::credential(subject, issuer, credType)))
+        return tecNO_ENTRY;
+
+    return tesSUCCESS;
+}
+
+TER
+CredentialDelete::deleteSLE(
+    ApplyView& view,
+    std::shared_ptr<SLE> const& sle,
+    beast::Journal j)
+{
+    if (!sle)
+        return tecNO_ENTRY;
+
+    AccountID const owner = sle->getAccountID(
+        (sle->getFlags() & lsfAccepted) ? sfSubject : sfIssuer);
+
+    auto const sleOwner = view.peek(keylet::account(owner));
+    if (!sleOwner)
+        return tecINTERNAL;
+
+    // Remove object from owner directory
+    std::uint64_t const page = sle->getFieldU64(sfOwnerNode);
+    if (!view.dirRemove(keylet::ownerDir(owner), page, sle->key(), false))
+    {
+        JLOG(j.fatal()) << "Unable to delete Credential from owner.";
+        return tefBAD_LEDGER;
+    }
+
+    adjustOwnerCount(view, sleOwner, -1, j);
+    view.update(sleOwner);
+
+    // Remove object from ledger
+    view.erase(sle);
+
+    return tesSUCCESS;
+}
+
+TER
+CredentialDelete::doApply()
+{
+    auto const subject = ctx_.tx[~sfSubject].value_or(account_);
+    auto const issuer = ctx_.tx[~sfIssuer].value_or(account_);
+
+    auto const credType(ctx_.tx[sfCredentialType]);
+    auto const sleCred =
+        view().peek(keylet::credential(subject, issuer, credType));
+
+    if ((subject != account_) && (issuer != account_))
+    {
+        if (credentialCheckNotExpired(
+                sleCred, ctx_.view().info().parentCloseTime))
+            return tecNO_PERMISSION;
+    }
+
+    return deleteSLE(view(), sleCred, j_);
+}
+
+// ------- APPLY --------------------------
+
+NotTEC
+CredentialAccept::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureCredentials))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    AccountID const issuer = ctx.tx[sfIssuer];
+    if (!issuer)
```

Comment:
This if statement will never trigger, because the getter above will error if the issuer doesn't exist (and the transaction won't even reach this point if the `sfIssuer` field isn't included).

Reply:
Please see Codecov
https://app.codecov.io/gh/XRPLF/rippled/pull/5103/blob/src/xrpld/app/tx/detail/Credentials.cpp